pandas/core/frame.py

```diff
@@ -1763,6 +1764,79 @@ def create_index(indexlist, namelist):
             columns = create_index(data["columns"], data["column_names"])
             return cls(realdata, index=index, columns=columns, dtype=dtype)
 
+    @classmethod
+    def from_pyarrow(
+        cls,
+        table,
+    ) -> DataFrame:
+        """
+        Convert a pyarrow table to DataFrame.
+
+        Parameters
+        ----------
+        table: pyarrow.Table
+
+        Returns
+        -------
+        DataFrame
+
+        See Also
+        --------
+        DataFrame.to_pyarrow
+        pyarrow.Table.to_pandas
+
+        Notes
+        -----
+        The conversion is zero-copy, and the resulting DataFrame uses
+        Arrow-backend data types.
+
+        For customization of the conversion use the
+        `to_pandas <https://arrow.apache.org/docs/python/generated/\
+pyarrow.Table.html#pyarrow.Table.to_pandas>`__
+        method of the Table object.
+
+        Examples
+        --------
+        >>> import pyarrow as pa
+        >>> table = pa.table([pa.array([1, 2, 3])], names=["my_column"])
+        >>> pd.DataFrame.from_pyarrow(table)
+           my_column
+        0          1
+        1          2
+        2          3
+        """
+        return table.to_pandas(types_mapper=ArrowDtype)
```

Comment:
you should do an instance check here ; eg if this is called with a non arrow table you would get a hard to understand error